presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoPageSource.java

```diff
@@ -183,7 +184,8 @@ else if (type.equals(DATE)) {
                     type.writeLong(output, TimeUnit.MILLISECONDS.toDays(utcMillis));
                 }
                 else if (type.equals(TIME)) {
-                    type.writeLong(output, UTC_CHRONOLOGY.millisOfDay().get(((Date) value).getTime()));
+                    int millis = UTC_CHRONOLOGY.millisOfDay().get(((Date) value).getTime());
+                    type.writeLong(output, millis * PICOSECONDS_PER_MILLISECOND);
```

Comment:
this can overlfow, because millis and PICOSECONDS_PER_MILLISECOND are both ints

also, this needs to round to `[0, PICOS_OF_DAY)`

Reply:
> also, this needs to round to [0, PICOS_OF_DAY)

`millis` comes from `millisOfDay`, so it has to be in the range `[0, MILLIS_PER_DAY]`. So `millis * PICOSECONDS_PER_MILLISECOND` is guaranteed to be in the range `[0, PICOS_OF_DAY)`

Reply:
you're correct. OTOH, you could save reader's time adorning this with a comment or doing `multiplyExact` -- just to consider